Server/string-decoder.ts

```diff
@@ -1,6 +1,14 @@
 export class StringDecoder {
     constructor() {}
+
     static toJSON(str: string) {
         return {};
     }
+
+    static hexToString(hex: string) {
+        let str: string = '';
+        for (let i = 0; (i < hex.length && hex.substr(i, 2) !== '00'); i += 2)
+            str += String.fromCharCode(parseInt(hex.substr(i, 2), 16));
```

Comment:
Add an if statement here to make sure the hex string only contains hex characters. If it detects a non-hex character in the input string, return null.

Also, typescript allows you to indicate the return type; do this:
```ts
static hexToString(hex: string): string {
```
If there is no return, use `void`. If you are unsure of the type, use `any`.